spec/requests/hosts_spec.rb

```diff
@@ -23,6 +23,17 @@
         expect(response).to have_http_status(:ok)
       end
 
+      it "prevents duplicate string/symbol keys mess" do
```

Comment:
@AlexanderZagaynov Perfect. I just wrote up the same spec and tested locally! So the spec LGTM
However I'd rather the title was a bit more descriptive. Perhaps:

>  it "can edit the password on a host when keys are specified as strings" do

Reply:
Or

it "can edit the password on a host when a new key is specified as strings" do

Reply:
@jvlcek I guess keys will always be a strings, because they came from request params. BTW, I tried to find what change caused this issue, but it seems pretty hard. Perhaps it is some Rails update.

Reply:
@jvlcek let me know if you still think to change description. I just can't say that your example is descriptive and precise, maybe we should write something else? :)

Reply:
@jvlcek missed your updates while was writing my response. Thing is that it is not about how keys are specified. It is about preventing situations like `{ 'userid' => 'a', :userid => 'b' }`. That's what I call "mess". Maybe this word confusing you, what about `it "prevents duplicate string/symbol keys mix" do`?

Reply:
done

Reply:
if we will stop using `.reverse_merge!` you can write nicely understandable  something like `it "updates userid  and password from params"`  ... 

 `it "updates only password from params..."`  ... 

`it "prevents duplicate string/symbol keys mess"` it looks like that it is related to some implementation detail or playing with keys in hash and there should be more real test case.

Reply:
@lpichler that `reverse_merge` was there before. And it is not the reason of the issue. Take a look at the `HashWithIndifferentAccess`, maybe it will help you to understand what is going on there. What about `reverse_merge` method complexity you found here - I guess that it is not normal to not understand the very basic features of the Rails framework.